### ISSUE TITLE
[ban] Optional explanation for banned globals

### ISSUE THREAD

**Issue by vsiao** [palantir/tslint#1940]
https://github.com/palantir/tslint/pull/1385 added the ability to give an optional explanation for banned functions. eg. instead of just:

```js
["_", "filter"]
// Calls to '_.filter' are not allowed.
```

you could provide an additional string to propose an alternative:

```js
["_", "filter", "Use the native JavaScript 'myArray.filter' instead."]
// Calls to '_.filter' are not allowed. Use the native JavaScript 'myArray.filter' instead.
```

https://github.com/palantir/tslint/pull/1408 added global bans, but left out the ability to provide an optional explanation. Obviously there's the problem of input syntax (if it's an array with 2 strings `"a"` and `"b"`, is that banning `a.b` or banning `a` with message `b`?), but I'd like to open a discussion for that.

**Comment by IllusionMH** [comment]
@vsiao as I can see in docs examples and [code](https://github.com/palantir/tslint/blob/master/src/rules/banRule.ts#L66-L67) - explanation isn't available for global bans.

Probably it is possible to check for characters that are not valid identifiers (spaces, special characters etc. since `ElementAccessExpression` aren't checked), or replace arrays with descriptive object
```json
{
    "path": ["object", "property"],
    "explanation": "Use native"
}
```

P.S. Is there any reason (except linted code became less readable) why `ElementAccessExpression` aren't checked in `ban` rule?
